lib/service_actor/nil_checkable.rb

```diff
@@ -8,6 +8,22 @@
 #     input :name, allow_nil: false
 #     output :user, allow_nil: false
 #   end
+#
+#   class CreateUser < Actor
+#     input :name,
+#           allow_nil: {
```

Comment:
The fact that this is no longer `false` in the "advanced mode" is a bit weird.

I don’t know what the best solution for this is. Perhaps this?:

```rb
allow_nil: {
  value: false,
  message: …
```


```rb
allow_nil: {
  allowed: false,
  message: …
```

Or, another way this could work would be to repeat the key, prefixed by `_message`. This strategy could be added to all other syntaxes. Although we are repeating somewhat the key, the fact that this does not introduce another nesting level makes it much simpler I think.

```rb
allow_nil: false,
allow_nil_message: -> _origin, _input, _klass do
   "The value #{input} cannot be empty"
end
```

What do you think?

Reply:
> The fact that this is no longer false in the "advanced mode" is a bit weird.

I agree. This is mistake.

> I don’t know what the best solution for this is. Perhaps this?:

The key `value` would look similar to the solution for `Defaultable`.

> Or, another way this could work would be to repeat the key, prefixed by _message. This strategy could be added to all other syntaxes. Although we are repeating somewhat the key, the fact that this does not introduce another nesting level makes it much simpler I think.

By the way, maybe that would be a really good option. In this case, by analogy, allow either String or Proc to be used.

Reply:
But this does not apply to `must`.

Reply:
I added an example for `NilCheckable`: https://github.com/sunny/actor/pull/82/commits/14d30db2eb300f5089f4158371f658ccacbba1f4

I propose to decide which version of the implementation is appropriate.